utils/service_discovery/sd_docker_backend.py

```diff
@@ -86,7 +137,33 @@ def _get_ports(self, container_inspect):
                     if co.get('name') == c_name:
                         ports = map(lambda x: str(x.get('containerPort')), co.get('ports', []))
         ports = sorted(ports, key=lambda x: int(x))
-        return ports
+        return self._extract_port_from_list(ports, tpl_var)
+
+    def _extract_port_from_list(self, ports, tpl_var):
+        if not ports:
+            return None
+
+        tpl_parts = tpl_var.split('_')
+
+        if len(tpl_parts) == 1:
+            log.debug("No index was passed for template variable %s. "
+                      "Trying with the last element." % tpl_var)
+            return ports[-1]
+
+        idx = self._is_int(tpl_parts[-1])
```

Comment:
you use `_is_int` just once.
this could be rewritten in:
```python
try:
    return ports[int(index)]
except ValueError:
    log.error("Index is not an integer. Using the last element instead")
except IndexError:
    log.error("Index is out of range. Using the last element instead")
return ports[-1]
```